packages/grid/backend/grid/core/node.py

```diff
@@ -15,10 +18,74 @@
 from grid.db.session import get_db_engine
 from grid.db.session import get_db_session
 
+
+def thread_function(*args, **kwargs) -> None:  # type: ignore
+    # TODO: call this after the Network node is deployed instead of using a timer.
+    time.sleep(30)
+
+    # third party
+    from requests import get
+
+    ip = get("https://api.ipify.org").content.decode("utf8")
```

Comment:
MAYBE: Andrew should we avoid depending on external services in our current ecosystem?

Reply:
As a general policy, definitely. In this case it's an attempt to get a public IP as one of multiple ways the system tries to establish a connection, and I'm not aware of a way to get a public IP reliably that doesn't involve connecting to an external service somehow. Thoughts?